### PR TITLE
feat(er/instance): support new parameter 'tags'

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The API has been support parameter 'tags' (CRD).

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new parameter: tags
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/er' TESTARGS='-run=TestAccInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run=TestAccInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccInstance_basic
=== PAUSE TestAccInstance_basic
=== CONT  TestAccInstance_basic
--- PASS: TestAccInstance_basic (96.71s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        96.815s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/er/resource_huaweicloud_er_instance_test.go [hunk 1]
```diff
@@ -91,13 +95,34 @@ func TestAccInstance_basic(t *testing.T) {
...
 
   name = "%[2]s"
   asn  = %[3]d
+
+  tags = {
+    foo = "bar"
+    key = "value"
+  }
+}
+`, acceptance.HW_AVAILABILITY_ZONE, name, bgpAsNum)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/er/resource_huaweicloud_er_instance_test.go:116]: Maybe add a preCheck for `HW_AVAILABILITY_ZONE`

File: huaweicloud/services/er/resource_huaweicloud_er_instance.go [hunk 2]
```diff
@@ -364,14 +407,75 @@ func resourceInstanceRead(ctx context.Context, d *schema.ResourceData, meta inte
...
+	deleteTagsPath = strings.ReplaceAll(deleteTagsPath, "{project_id}", client.ProjectID)
+	deleteTagsPath = strings.ReplaceAll(deleteTagsPath, "{resource_type}", resourceType)
+	deleteTagsPath = strings.ReplaceAll(deleteTagsPath, "{resource_id}", resourceId)
+	deleteTagsPath = strings.ReplaceAll(deleteTagsPath, "{key}", key)
+
+	deleteTagsOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			204,
+		},
```
Comment (deer-hang) [inline huaweicloud/services/er/resource_huaweicloud_er_instance.go]: For delete method, do not need write `OkCodes` 204

File: huaweicloud/services/er/resource_huaweicloud_er_instance.go [hunk 3]
```diff
@@ -364,14 +407,75 @@ func resourceInstanceRead(ctx context.Context, d *schema.ResourceData, meta inte
...
+	return nil
+}
+
+func updateResourceTags(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	oRaw, nRaw := d.GetChange("tags")
+	oMap := oRaw.(map[string]interface{})
+	nMap := nRaw.(map[string]interface{})
+	instanceId := d.Id()
+
+	if len(oMap) > 0 {
```
Comment (deer-hang) [inline huaweicloud/services/er/resource_huaweicloud_er_instance.go]: It seems like, this line is unnecessary

File: huaweicloud/services/er/resource_huaweicloud_er_instance.go [hunk 4]
```diff
@@ -213,6 +247,15 @@ func resourceInstanceCreate(ctx context.Context, d *schema.ResourceData, meta in
 		return diag.FromErr(err)
 	}
 
+	if tags, ok := d.GetOk("tags"); ok {
+		for k, v := range tags.(map[string]interface{}) {
+			err := addResourceTags(createInstanceClient, "instance", id.(string), k, v.(string))
+			if err != nil {
+				return diag.Errorf("error setting tags to the ER instance: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/er/resource_huaweicloud_er_instance.go]: The err message from function addResourceTags has already been defined, you can just return diag.FromErr(err)

File: huaweicloud/services/acceptance/er/resource_huaweicloud_er_instance_test.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/er/resource_huaweicloud_er_instance_test.go:116]: using data source instead.

File: huaweicloud/services/er/resource_huaweicloud_er_instance.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/er/resource_huaweicloud_er_instance.go]: sure and fixed.

File: huaweicloud/services/er/resource_huaweicloud_er_instance.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/er/resource_huaweicloud_er_instance.go]: yeah
already removed.

File: huaweicloud/services/er/resource_huaweicloud_er_instance.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/er/resource_huaweicloud_er_instance.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/er' TESTARGS='-run=TestAccInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run=TestAccInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccInstance_basic
=== PAUSE TestAccInstance_basic
=== CONT  TestAccInstance_basic
--- PASS: TestAccInstance_basic (59.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        59.935s
```

**Pull request merged**